server/src/main/java/org/elasticsearch/action/get/MultiGetShardRequest.java

```diff
@@ -174,6 +174,14 @@ void add(int location, MultiGetRequest.Item item) {
         this.items.add(item);
     }
 
+    public List<Integer> locations() {
```

Comment:
I think instead of exposing these two function, you can expose a `getNumberOfItems()` function, and that'll be all you need.

Reply:
Unfortunately, we do need expose locations for `MultiGetShardResponse`, but exposing items in indeed unnecessary

Reply:
This change may be reverted with the new approach we took.

Reply:
Reverted in https://github.com/elastic/elasticsearch/commit/ed8c90206a9b1e57a5b65f861577350fbb5b4c45